File: app/src/main/java/com/kickstarter/ui/viewholders/ProjectSocialViewHolder.java
package com.kickstarter.ui.viewholders;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.kickstarter.R;
import com.kickstarter.libs.transformations.CircleTransformation;
import com.kickstarter.models.User;
import com.squareup.picasso.Picasso;

import butterknife.Bind;
import butterknife.ButterKnife;

import static com.kickstarter.libs.utils.ObjectUtils.requireNonNull;

public class ProjectSocialViewHolder extends KSViewHolder {
  private User user;
  protected @Bind(R.id.friend_image) ImageView friendImageView;
  protected @Bind(R.id.friend_name) TextView friendNameTextView;

  public ProjectSocialViewHolder(final @NonNull View view) {
    super(view);
    ButterKnife.bind(this, view);
  }

  @Override
  public void bindData(final @Nullable Object data) throws Exception {
    this.user = requireNonNull((User) data, User.class);
  }

  @Override
  public void onBind() {
    Picasso.get().load(this.user.avatar().small())
    .transform(new CircleTransformation())
    .into(this.friendImageView);

    this.friendNameTextView.setText(this.user.name());
  }
}

File: app/src/main/java/com/kickstarter/ui/viewholders/ProjectSocialViewHolder.kt
package com.kickstarter.ui.viewholders

import com.kickstarter.databinding.ProjectSocialViewBinding
import com.kickstarter.libs.utils.extensions.addToDisposable
import com.kickstarter.libs.utils.extensions.isNotNull
import com.kickstarter.models.User
import com.kickstarter.ui.extensions.loadCircleImage
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.subjects.BehaviorSubject

class ProjectSocialViewHolder(private val binding: ProjectSocialViewBinding) : KSViewHolder(binding.root) {
    private var friendObserver = BehaviorSubject.create<User>()
    private val disposables = CompositeDisposable()
    @Throws(Exception::class)
    override fun bindData(data: Any?) {
        val user = requireNotNull(data as User?) { User::class.java.toString() + " required to be non-null." }
        if (user.isNotNull()) friendObserver.onNext(user)
    }

    init {
        friendObserver
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe {
                binding.friendImage.loadCircleImage(it.avatar().medium())
                binding.friendName.text = it?.name()
            }
            .addToDisposable(disposables)
    }

    override fun destroy() {
        disposables.clear()
        super.destroy()
    }
}
